armi/reactor/blocks.py

```diff
@@ -1929,6 +1938,48 @@ def _getPinCoordinatesHex(self):
                 coordinates.append(xyz)
         return coordinates
 
+    def autoCreateSpatialGrids(self):
+        """
+        Given a block without a spatialGrid, create a spatialGrid and give its children
+        the corresponding spatialLocators (if it is a simple block). In this case, a simple block would
+        be one that has either multiplicity of components equal to 1 or n but no other multiplicities.
+        Otherwise, raise a ValueError since the number of multiplicities makes it so we cannot generate
+        a grid for this block.
+
+        In other words, here we gather all components to either be a multiIndexLocation containing all
+        of the pin positions, otherwise, locator is the center (0,0).
+        """
+
+        # Check multiplicities...
+        mults = {c.getDimension("mult") for c in self}
+
+        if len(mults) != 2 or 1 not in mults:
+            self.spatialGrid = None
+            raise ValueError(
+                "Could not create a spatialGrid for block {}, multiplicities are not 1 or N they are {}".format(
+                    self.p.type, mults
+                )
+            )
+
+        spatialLocators = grids.MultiIndexLocation(grid=self.spatialGrid)
+        ringNumber = hexagon.numRingsToHoldNumCells(self.getNumPins())
```

Comment:
Check that the number of locations in `ringNumber` rings is the same as the number of pins